array_api_compat/cupy/_aliases.py

```diff
@@ -69,20 +67,21 @@
 # asarray also adds the copy keyword, which is not present in numpy 1.0.
 def asarray(
     obj: Union[
-        ndarray,
+        Array,
         bool,
         int,
         float,
-        NestedSequence[bool | int | float],
+        complex,
+        NestedSequence[bool | int | float | complex],
         SupportsBufferProtocol,
     ],
```

Comment:
if I remember correctly, arrays are required to support the buffer protocol, so you could simplify this to:

```
    obj: float | NestedSequence[float] | SupportsBufferProtocol,
```

Reply:
Do you mean `obj: complex | NestedSequence[complex] | SupportsBufferProtocol`?
I do worry about

Reply:
oh woops, yea, good catch